docs/spec.md

```diff
@@ -1647,6 +1647,68 @@ for this operation ([#560](https://github.com/openxla/stablehlo/issues/560)).
 
 &nbsp;[More Examples](../stablehlo/tests/interpret_clamp.mlir)
 
+### collective_broadcast
```

Comment:
@ghpvnist could you give a first-pass review on the spec. Anything missing from the checklist?

Reply:
Also take a look at the verification and type inference code. Interested in any stylistic suggestions to ensure good readability / correlation with the spec.